vshard/replicaset.lua

```diff
@@ -717,7 +717,7 @@ local function replicaset_map_call(replicaset, func, args, opts)
     --
     map = table.new(0, replica_count)
     for uuid, future in pairs(futures) do
-        res, err = future:wait_result(timeout)
+        res, err = util.future_wait(future, timeout)
```

Comment:
Lets add this `local future_wait = util.future_wait` to the beginning of the file (same place where `fiber_cond_wait` is saved) and use this `future_wait()` everywhere. The reason is that in Lua a table index operation like `table.key` is not free, it is computed at runtime on each use.

Same in the router.